frontend/exporter/src/types/copied.rs

```diff
@@ -2982,6 +2983,25 @@ pub struct AnonConst<Body: IsBody> {
     pub body: Body,
 }
 
+/// Reflects [`rustc_hir::ConstArg`]
+#[derive_group(Serializers)]
+#[derive(AdtInto, Clone, Debug, JsonSchema)]
+#[args(<'tcx, S: UnderOwnerState<'tcx>>, from: rustc_hir::ConstArg<'tcx>, state: S as s)]
+pub struct ConstArg<Body: IsBody> {
+    pub hir_id: HirId,
+    pub kind: ConstArgKind<Body>,
+    pub is_desugared_from_effects: bool,
+}
+
+/// Reflects [`rustc_hir::ConstArgKind`]
+#[derive_group(Serializers)]
+#[derive(AdtInto, Clone, Debug, JsonSchema)]
+#[args(<'tcx, S: UnderOwnerState<'tcx>>, from: rustc_hir::ConstArgKind<'tcx>, state: S as s)]
+pub enum ConstArgKind<Body: IsBody> {
+    Path(QPath),
+    Anon(AnonConst<Body>),
+}
```

Comment:
We could get rid of all that by using `ty::GenericParamDefKind` instead of `hir::GenericParamKind`. I'm generally in favor of using things in `ty` rather then their HIR equivalent since the `ty` stuff is more processed.